src/particles/beam/BeamParticleContainer.H

```diff
@@ -217,35 +210,52 @@ public:
     int m_insitu_period {0};
 private:
     std::string m_name; /**< name of the species */
+    /** injection type, fixed_width or fixed_ppc */
+    std::string m_injection_type;
+
+    // fixed_ppc:
+
+    amrex::IntVect m_ppc {1, 1, 1}; /**< Number of particles per cell in each direction */
     amrex::Real m_zmin; /**< Min longitudinal particle position of the beam */
     amrex::Real m_zmax; /**< Max longitudinal particle position of the beam */
     amrex::Real m_radius; /**< Radius of the can beam */
-    amrex::IntVect m_ppc {1, 1, 1}; /**< Number of particles per cell in each direction */
+    amrex::RealVect m_position_mean {0., 0., 0.}; /**< mean position of the beam */
+    amrex::Real m_min_density {0.}; /**< minimum density at which beam particles are generated */
+    amrex::IntVect m_random_ppc {0, 0, 0}; /**< if the cell position is random in each direction */
+    GetInitialDensity m_get_density {}; /**< density profile of the beam */
+    GetInitialMomentum m_get_momentum {}; /**< momentum profile of the beam */
+    /** Density parser for fixed-ppc beam. Owns data for m_density_func */
+    amrex::Parser m_density_parser;
+    uint64_t m_id64 = 1; /**< 64 bit ID to initialize many particles without overflowing */
+
+    // fixed_weight:
+
     /** Average x position of the fixed-weight beam depending on z */
     amrex::Parser m_pos_mean_x_parser;
     /** Average y position of the fixed-weight beam depending on z */
     amrex::Parser m_pos_mean_y_parser;
-    /** Density parser for fixed-ppc beam. Owns data for m_density_func */
-    amrex::Parser m_density_parser;
     /** Width of the Gaussian beam. Only used for a fixed-weight beam */
     amrex::RealVect m_position_std {0., 0., 0.};
```

Comment:
This has nothing to do with this PR, but isn't that also used for a fixed ppc beam with a Gaussian profile?

Reply:
Position std for the fixed ppc Gaussian profile is stored inside `GetInitialDensity`, so that variable in `BeamParticleContainer` is only used for fixed weight beams.

Reply:
Ah, I see, that makes sense